read-fonts/src/tables/varc.rs

```diff
@@ -152,55 +148,36 @@ impl<'a> VarcComponent<'a> {
             None
         };
 
-        let dx = if flags.contains(VarcFlags::HAVE_TRANSLATE_X) {
-            cursor.read::<FWord>()?.to_i16()
-        } else {
-            0
-        };
-        let dy = if flags.contains(VarcFlags::HAVE_TRANSLATE_Y) {
-            cursor.read::<FWord>()?.to_i16()
-        } else {
-            0
-        };
-
-        let rotation = if flags.contains(VarcFlags::HAVE_ROTATION) {
-            cursor.read::<F4Dot12>()?.to_f32()
-        } else {
-            0.0
-        };
-
-        let sx = if flags.contains(VarcFlags::HAVE_SCALE_X) {
-            cursor.read::<F6Dot10>()?.to_f32()
-        } else {
-            1.0
-        };
-        let sy = if flags.contains(VarcFlags::HAVE_SCALE_Y) {
-            cursor.read::<F6Dot10>()?.to_f32()
-        } else {
-            sx
-        };
-
-        let skewx = if flags.contains(VarcFlags::HAVE_SKEW_X) {
-            cursor.read::<F4Dot12>()?.to_f32()
-        } else {
-            0.0
-        };
-        let skewy = if flags.contains(VarcFlags::HAVE_SKEW_Y) {
-            cursor.read::<F4Dot12>()?.to_f32()
-        } else {
-            0.0
-        };
-
-        let center_x = if flags.contains(VarcFlags::HAVE_TCENTER_X) {
-            cursor.read::<FWord>()?.to_i16()
-        } else {
-            0
-        };
-        let center_y = if flags.contains(VarcFlags::HAVE_TCENTER_Y) {
-            cursor.read::<FWord>()?.to_i16()
+        let mut transform = DecomposedTransform::default();
```

Comment:
does it make sense to now just have a constructor on this type that takes the cursor and the flags and parses itself?

Reply:
My rationale for the current structure was to keep the bytestream parsing together. I would like to keep that unless you feel strongly about it. If so I will address in a follow-on.